x/harvest/keeper/liquidation.go

```diff
@@ -253,6 +253,81 @@ func (k Keeper) StartAuctions(ctx sdk.Context, borrower sdk.AccAddress, borrows,
 	return nil
 }
 
+// GetCurrentLTV calculates the user's current LTV based on their deposits/borrows in the store
+func (k Keeper) GetCurrentLTV(ctx sdk.Context, addr sdk.AccAddress) (sdk.Dec, error) {
+	// Fetch deposits and parse coin denoms
+	deposits := k.GetDepositsByUser(ctx, addr)
+	depositDenoms := []string{}
+	for _, deposit := range deposits {
+		depositDenoms = append(depositDenoms, deposit.Amount.Denom)
+	}
+
+	// Fetch borrow balances and parse coin denoms
+	borrowBalances := k.GetBorrowBalance(ctx, addr)
+	if borrowBalances.IsZero() {
+		return sdk.ZeroDec(), nil
+	}
+	borrowDenoms := getDenoms(borrowBalances)
+
+	liqMap := make(map[string]LiqData)
+
+	// Load required liquidation data for every deposit/borrow denom
+	denoms := removeDuplicates(borrowDenoms, depositDenoms)
+	for _, denom := range denoms {
+		mm, found := k.GetMoneyMarket(ctx, denom)
+		if !found {
+			return sdk.ZeroDec(), sdkerrors.Wrapf(types.ErrMarketNotFound, "no market found for denom %s", denom)
+		}
+
+		priceData, err := k.pricefeedKeeper.GetCurrentPrice(ctx, mm.SpotMarketID)
+		if err != nil {
+			return sdk.ZeroDec(), err
+		}
+
+		liqMap[denom] = LiqData{priceData.Price, mm.BorrowLimit.LoanToValue, mm.ConversionFactor}
+	}
+
+	// Build valuation map to hold deposit coin USD valuations
+	depositCoinValues := types.NewValuationMap()
+	for _, deposit := range deposits {
+		dData := liqMap[deposit.Amount.Denom]
+		dCoinUsdValue := sdk.NewDecFromInt(deposit.Amount.Amount).Quo(sdk.NewDecFromInt(dData.conversionFactor)).Mul(dData.price)
+		depositCoinValues.Increment(deposit.Amount.Denom, dCoinUsdValue)
+	}
+
+	// Build valuation map to hold borrow coin USD valuations
+	borrowCoinValues := types.NewValuationMap()
+	for _, bCoin := range borrowBalances {
+		bData := liqMap[bCoin.Denom]
+		bCoinUsdValue := sdk.NewDecFromInt(bCoin.Amount).Quo(sdk.NewDecFromInt(bData.conversionFactor)).Mul(bData.price)
+		borrowCoinValues.Increment(bCoin.Denom, bCoinUsdValue)
+	}
+
+	// User doesn't have any deposits, catch divide by 0 error
+	sumDeposits := depositCoinValues.Sum()
+	if sumDeposits.Equal(sdk.ZeroDec()) {
+		return sdk.ZeroDec(), nil
+	}
+
+	// Loan-to-Value ratio
+	return borrowCoinValues.Sum().Quo(sumDeposits), nil
+}
+
+// UpdateItemInLtvIndex updates the key a borrower's address is stored under in the LTV index
+func (k Keeper) UpdateItemInLtvIndex(ctx sdk.Context, prevLtv sdk.Dec, borrower sdk.AccAddress) error {
+	currLtv, err := k.GetCurrentLTV(ctx, borrower)
+	if err != nil {
+		return err
+	}
+
+	k.RemoveFromLtvIndex(ctx, prevLtv, borrower)
+	// If the user doesn't have any borrows their LTV is 0
+	if currLtv.GT(sdk.ZeroDec()) {
```

Comment:
Question: Is it possible for a user to have a micro borrow where the ltv rounds to zero and thus we don't store it in the index? If that happened, could it cause a panic/unexpected error somewhere?

Reply:
Good point. I updated `GetCurrentLTV` with a bool that indicates if the LTV should be added to the index instead of using `sdk.ZeroDec()` as a bool for this conditional in https://github.com/Kava-Labs/kava/pull/742/commits/8c2bb779ad0a7b7efa681dd12753efa77ad5de54.

Additionally, I'm using the bool returned by the function when getting `prevLtv` to determine if the borrower should be removed from the index, because only existing items in the index should be removed. This would have failed silently without issues because attempting to remove a non-existent item from the index doesn't return an error, but probably still better to prevent it from happening.